internal/worker/api/server_test.go

```diff
@@ -0,0 +1,559 @@
+package api
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/renatoaguimaraes/job-scheduler/internal/worker/proto"
+	"github.com/renatoaguimaraes/job-scheduler/pkg/worker/conf"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
+)
+
+var config = conf.Config{ServerAddress: "localhost:8080", LogFolder: os.TempDir()}
+
+func TestStartAuthnAuthzAdminUser(t *testing.T) {
+	// load server credentials
+	servercred, err := loadServerCredentials(clientca, servercert, serverkey)
+	assert.Nil(t, err)
```

Comment:
here and below: use `require` instead of `assert` (different package from testify) to terminate the test on a failure; also use the appropriate helper functions for the check
```suggestion
	require.NoError(t, err)
```

Reply:
Thanks. Good tip.